pkgs/test_core/lib/src/runner.dart

```diff
@@ -106,6 +107,12 @@ class Runner {
               '`DART_VM_OPTIONS=-DSILENT_OBSERVATORY=true`.');
         }
 
+        if (_config.coverage != null) {
+          var directory =
+              await Directory(_config.coverage).create(recursive: true);
+          print('Gathering coverage and outputting to: ${directory.path}');
```

Comment:
[nit] I'm guessing this was just here for debugging and should be removed before merge?

Reply:
I think I'd want some logging to confirm that coverage is being gathered but that might just be me

Reply:
In that case, I'm guessing there is a more robust way to log that doesn't break reporter contracts like the JSON reporter.

Reply:
Fixed!